pkg/archive/time.go

```diff
@@ -0,0 +1,12 @@
+package archive // import "github.com/docker/docker/pkg/archive"
+
+import "time"
+
+const maxSeconds = (1<<63 - 1) / int64(1e9)
+
+func sanitizeTime(t time.Time) time.Time {
+	if ut := t.Unix(); ut < 0 || ut > maxSeconds {
+		return time.Unix(0, 0)
+	}
+	return t
+}
```

Comment:
Just to make sure I blame myself not asking; I recall there was some special handling for 32-bit; is that no longer needed, or was that only needed for Windows situations? https://github.com/moby/moby/blob/bdfc38435245cf052f247d4cc0b1f9e9b71909c9/pkg/system/chtimes.go#L10-L27

Reply:
I'm just going to pull in containerd's implementation for this, at some point we should have this unified